changelog/29974.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:change
+server: disable_mlock configuration option is now required for integrated storage and no longer has a default. If you are using the default value with integrated storage, you must now explicitly set disable_mlock to true or false.
```

Comment:
```suggestion
server: disable_mlock configuration option is now required for integrated storage and no longer has a default. If you are using the default value with integrated storage, you must now explicitly set disable_mlock to true or false or Vault server will fail to start.
```

Should we call out the consequences more explicitly? I think the message was clear enough before, but I would like us to really drive this point home 😄